model/clusters_mgmt/v1/cluster_image_registry_type.model

```diff
@@ -0,0 +1,4 @@
+struct ClusterImageRegistry {
+	ControlPlaneIdentity AzureControlPlaneManagedIdentity
```

Comment:
Another related question we have here is: what happens if the cluster image registry functionality is not supported only on azure but also on aws. We would need to have here something like:
```
ClusterImageRegistry:
AWS:
  ControlPlaneIdentity AWSSTSIdentity
  DataPlaneIdentity AWSSTSIdentity
Azure:
  ControlPlaneIdentity AzureControlPlaneManagedIdentity
  DataPlaneIdentity AzureDataPlaneManagedIdentity
```

or similar

Reply:
This is an interesting problem. For completion, the way we've solved this in ROSA is by providing an array of arbitrary operators that require identity:
```
{
  "aws": {
    "operator_iam_roles": [
      {"id": "openshift-image-registry-operator", "role_arn": "<identity resource>"},
    ]
  }
}
```
However, when it comes to any knobs for said operator, they don't live there, they are provided elsewhere on an as-needed basis.

We should probably follow a similar model.

Reply:
A similar approach is what's implemented in master currently. The difference is that instead of having an array we have a free-form map where the key is the id of the operator.

I explain the detail about that here https://github.com/openshift-online/ocm-api-model/pull/1037#discussion_r2092609890

The use of a map instead of an array is *intentional*. The reason is that arrays are not PATCH friendly. JSON Merge Patch doesn’t support adding, removing, or changing array elements (except by explicitly replacing the whole array).

Reply:
I've pushed an update on this in line with what Miguel was suggesting above.